Restore check for `sectionListRender`

MarmadileManteater opened the pull request:
# Restore check for `sectionListRender`

<!-- Thanks for sending a pull request! Make sure to follow the contributing guidelines. -->
<!-- Important note, we may remove your pull request if you do not use this provided PR template correctly. -->

## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->
- [X] Bugfix
- [ ] Feature Implementation
- [ ] Documentation
- [ ] Other


## Description
Music channels such as [Infected Mushroom](https://www.youtube.com/channel/UCrbvoMC0zUvPL8vjswhLOSw) still use the `gridRenderer` to display their content, and as a result, `yt-channel-info` is returning no videos for these channels. This PR restores the check for the `sectionListRenderer` in order to fix this issue with music channels.

## Screenshots <!-- If appropriate -->
_before:_
![image](https://user-images.githubusercontent.com/106682128/199768392-7cf9cf7a-6e2b-4426-8a05-0412b30e3684.png)
_after:_
![image](https://user-images.githubusercontent.com/106682128/199767611-1d2adf80-92a2-4ffd-9545-f585d70ab89d.png)

## Testing <!-- for code that is not small enough to be easily understandable -->
Here are some examples of channels which qualify as "music" channels if you want to check them:
- [Linkin Park](https://www.youtube.com/channel/UCZU9T1ceaOgwfLRq7OKFU4Q)
- [My Chemical Romance](https://www.youtube.com/channel/UCCZGYab5SpD0I7Z5JqJZgww)
- [beegees](https://www.youtube.com/channel/UCD9sCcKXnFxMeuFoNayVxeQ)

## Additional Information
It might be worth adding a test which checks a music channel in order to make sure the response is not empty.

This issue was originally brought up by someone on the freetube matrix channel.

MarmadileManteater committed:
Restore check for `sectionListRender`

Music channels still use the `gridRenderer`.

ChunkyProgrammer commented:
Could you fix the linter issue and it might be a good idea to replace shorts test with a test for one of the music channels :)

MarmadileManteater committed:
Fix accidental duplicate branch condition

I accidentally added the same condition twice to an `if else`.
The result was that one of them would never execute.

MarmadileManteater commented:
> Could you fix the linter issue and it might be a good idea to replace shorts test with a test for one of the music channels :)

The linter caught a conditional branch which I had accidentally made impossible to reach.

MarmadileManteater committed:
Adjust shorts test

Shorts don't display a visible lengthSeconds on channel pages
at the moment, and I am unsure where this data could be pulled from
without making additional calls.

MarmadileManteater commented:
If we are going to get rid of the shorts test, we could just adjust it to be more accurate to what is currently working. Looking at the other tests in the same file, most of them expect `data.items.length` to be zero. If the shorts channel test was changed to expect the same thing as the other channel video tests, the test would work, and it would still catch if shorts channels suddenly started returning nothing.

MarmadileManteater committed:
Add music channel test
